### PR TITLE
Add Kubernetes example

### PR DESCRIPTION
**Pull request by dchw**
This is enough to demo getting it up; with any Earthly-specific stuff configured. Going much further would be implementation specific; or add too much noise.

### CODE REVIEW

File: kubernetes/readme.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Running Earthly in Kubernetes
+
+This is an example ddeployment of Earthly in Kubernetes. It uses an external `buildkitd` service to allow better isolation of the privileged pods.
```
Comment (vladaionescu) [inline kubernetes/readme.md]:
```suggestion
This is an example deployment of Earthly in Kubernetes. It uses an external `buildkitd` service to allow better isolation of the privileged pods.
```

**Review by vladaionescu — APPROVED** [review]
Looks great - is there a way to automatically test this setup as part of a CI run to ensure it doesn't break over time for some reason?

**Review by alexcb — APPROVED** [review]
woohoo!

**Comment by dchw** [comment]
> Looks great - is there a way to automatically test this setup as part of a CI run to ensure it doesn't break over time for some reason?

:thinking: I would love to get this (and Jenkins, and GitLab) under CI. We should do it, 100%.

But most of these require a fair bit of local setup (see the `LOCALLY` stuff in each examples `Earthlfile`). I would prefer to grab a low-cost instance somewhere and isolate most of this local setup. I also don't think it needs to run on each PR; this feels like one of those every night kind of deals.

*Edit:* Opened an issue for this: https://github.com/earthly/ci-examples/issues/2

**Pull request merged**